GitUpKit/Core/GCRepository+Reflog.m

```diff
@@ -66,17 +62,15 @@ - (id)initWithRepository:(GCRepository*)repository entry:(const git_reflog_entry
 }
 
 - (void)dealloc {
-  [_fromSHA1 release];
-  [_fromCommit release];
-  [_toSHA1 release];
-  [_toCommit release];
-  [_committerName release];
-  [_committerEmail release];
-
-  [_references release];
-  [_messages release];
-
-  [super dealloc];
+  _fromSHA1 = nil;
```

Comment:
Let's remove all the `nil` assignments. It's just extra code that doesn't need to be there. Please apply this suggestion to everywhere in this PR. Thanks!

Reply:
@lucasderraugh 
Fixed.
Could you also comment on #858?
And I noticed that you merged your own PR with new libgit2. 
Can we now check and merge File history PR?